packages/entities/entities-plugins/src/components/PluginSelectCard.vue

```diff
@@ -0,0 +1,310 @@
+        #actions
+      >
+        <KDropdownMenu
+          :kpop-attributes="{ placement: 'bottom-end' }"
+          width="150"
+        >
+          <template #default>
+            <KButton
+              class="actions-trigger"
+              data-testid="overflow-actions-button"
+              size="small"
+            >
+              <template #icon>
+                <KIcon
+                  :color="KUI_COLOR_TEXT_NEUTRAL_STRONGER"
+                  icon="more"
+                  :size="KUI_ICON_SIZE_30"
+                />
+              </template>
+            </KButton>
+          </template>
+
+          <template #items>
+            <KDropdownItem
+              v-if="canDeleteCustom"
+              data-testid="edit-plugin-schema"
+              @click.stop="handleCustomEdit(plugin.name)"
+            >
+              {{ t('actions.edit') }}
+            </KDropdownItem>
+            <KDropdownItem
+              v-if="canDeleteCustom"
+              data-testid="delete-plugin-schema"
+              has-divider
+              is-dangerous
+              @click.stop="handleCustomDelete"
+            >
+              {{ t('actions.delete') }}
+            </KDropdownItem>
+          </template>
+        </KDropdownMenu>
+      </template>
+
+      <template #body>
+        <div
+          class="plugin-card-body"
+          :class="{ 'custom-plugin': isCustomPlugin }"
+          :title="!plugin.available ? t('plugins.select.unavailable_tooltip') : plugin.name"
+          @click="handleCustomClick"
+        >
+          <h4 class="plugin-card-title">
+            {{ plugin.name }}
+          </h4>
+          <PluginIcon
+            :alt="plugin.name"
+            class="plugin-card-icon"
+            :name="plugin.imageName || plugin.id"
+            :size="55"
+          />
+          <p
+            v-if="plugin.description"
+            class="plugin-card-text"
+          >
+            {{ plugin.description }}
+          </p>
+        </div>
+        <div
+          :class="{
+            'plugin-card-create-footer': isCreateCustomPlugin,
+            'plugin-card-footer': !isCreateCustomPlugin,
+          }"
+          @click="handleCustomClick"
+        >
+          {{ isCreateCustomPlugin ? t('actions.create_custom') : plugin.exists ? t('actions.enabled') : t('actions.enable') }}
+        </div>
+      </template>
+    </KCard>
+  </KTooltip>
+</template>
+
+<script setup lang="ts">
+import { computed, type PropType } from 'vue'
+import { useRouter } from 'vue-router'
+import { PluginGroup, type KongManagerPluginFormConfig, type KonnectPluginFormConfig, type PluginType } from '../types'
+import { KUI_ICON_SIZE_30, KUI_COLOR_TEXT_NEUTRAL_STRONGER } from '@kong/design-tokens'
+import composables from '../composables'
+import PluginIcon from './PluginIcon.vue'
+
+const emit = defineEmits<{
+  (e: 'plugin-clicked', plugin: PluginType) : void,
+  (e: 'custom-plugin-delete'): void,
```

Comment:
I didn't see this event documented anywhere (sorry if I missed it)?

Reply:
This is a child component that we don't expose to consuming apps. The emit is only used by `CustomPluginGrid` which is also not exposed to consuming apps.

Reply:
Same suggestion here; can you just add a comment about what the event is for and that it's internal-only?